### PR TITLE
Rollup 1.0 + other updates

### PR DESCRIPTION
**Pull request by ForsakenHarmony**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -523,7 +523,7 @@ function createConfig(options, entry, format, writeMeta) {
 					useTypescript &&
 						typescript({
 							typescript: require('typescript'),
-							cacheRoot: `./.rts2_cache_${format}`,
+							cacheRoot: `./node_modules/.rts2_cache_${format}`,
```
Comment (developit) [inline src/index.js]: clever.

Comment (maraisr) [reply src/index.js]: Rekon we could take that 1 step further with,

```suggestion
							cacheRoot: `./node_modules/.rts2_cache_${format}`,
							cacheRoot: `./node_modules/.cache/.rts2_cache_${format}`,
```

File: package.json [hunk 2]
```diff
@@ -42,61 +42,61 @@
...
-    "rollup-plugin-terser": "^3.0.0",
-    "rollup-plugin-typescript2": "^0.19.0",
+    "rollup-plugin-postcss": "^2.0.3",
+    "rollup-plugin-terser": "^4.0.4",
+    "rollup-plugin-typescript2": "^0.20.1",
     "sade": "^1.4.0",
     "tiny-glob": "^0.2.6",
     "tslib": "^1.9.0",
-    "typescript": ">=2.8.3"
+    "typescript": "^3.3.4000"
```
Comment (maraisr) [inline package.json]: Now that [TypeScript 3.4](https://github.com/Microsoft/TypeScript/releases/tag/v3.4.1) is a thing, do you think we could make that read;

```suggestion
    "typescript": "^3.4.0"
```

File: src/index.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import fs from 'fs';
 import { resolve, relative, dirname, basename, extname } from 'path';
-import chalk from 'chalk';
+import { green, red, yellow, white, blue } from 'kleur';
```
Comment (maraisr) [inline src/index.js]: :100:

File: src/index.js (hunk 1 above)
Comment (ForsakenHarmony) [reply src/index.js]: was kinda annoyed with those 3 directories in my project dir 🙃

File: src/index.js [hunk 4]
```diff
@@ -219,17 +225,19 @@ export default async function microbundle(inputOptions) {
...
+		steps.map(config => async () => {
+			const { inputOptions, outputOptions } = config;
 			inputOptions.cache = cache;
 			let bundle = await rollup(inputOptions);
 			cache = bundle;
-			const { code } = await bundle.write(outputOptions);
-			return await getSizeInfo(code, outputOptions.file);
+			await bundle.write(outputOptions);
+			return await config._sizeInfo;
+			// return await getSizeInfo(code, outputOptions.file, options.raw);
```
Comment (maraisr) [inline src/index.js]: You've commented here, why if you don't mind me asking?

Comment (ForsakenHarmony) [reply src/index.js]: oh that should be removed

File: src/index.js [hunk 5]
```diff
@@ -16,7 +16,6 @@ import postcss from 'rollup-plugin-postcss';
 import gzipSize from 'gzip-size';
 import brotliSize from 'brotli-size';
 import prettyBytes from 'pretty-bytes';
-import shebangPlugin from 'rollup-plugin-preserve-shebang';
```
Comment (maraisr) [inline src/index.js]: Oh good - I seriously spent about 30 mins trying to PR that project with updates. And I could not get rollup to like anything I was doing in that `transformChunk` hook. 👀

**Comment by ForsakenHarmony** [comment]
Oh yes, seems like I forgot to push that commit

**Pull request merged**